.github/workflows/ci.yml

```diff
@@ -113,7 +114,7 @@ jobs:
         run: |
           tox ${{ needs.check_mturk_changes.outputs.needs_mturk_tests == 'true' && '-- --mturkfull' || ''}}
           npm run test --coverage
-        if: matrix.python-version == 3.11
```

Comment:
I don't really see the logic of this and the below `python-version` bits. I think the logic is that we should only run certain 'full' tests on the latest version of Python; we shouldn't run them on every version above and including 3.11. I think we should just be replacing `3.11` with `3.12` and leaving everything else the same, no?

Reply:
OK, I thought it might be more secure to run the full tests on the two latest versions as there is not much additional cost and we can be sure that people still on the recent latest version of Python (in this case 3.11) are assured the are no issues with code changes that had to be made for the Python version upgrade.

Reply:
See for example [this commit](https://github.com/Dallinger/Dallinger/pull/6013/commits/5ceff4150986ac1e22233a1ea6efc7ab1b61e035) which required a code change to cope with breaking API change: 

The best would probably be to run the full tests on all versions. But i didn't want to go that far.